### ISSUE TITLE
Use ui theme definitions for colors

### ISSUE THREAD

**Issue by parkerault** [basyura/inkdrop-sidetoc#4]
When using a dark UI theme the highlight color makes the text unreadable.

I suggest using the `@sidebarMenuItemActiveBackground` variable instead of defining a custom plugin setting.

You can see the problem here:
![image](https://user-images.githubusercontent.com/78856/92774734-603ace80-f363-11ea-9581-524b5f740d40.png)

**Comment by basyura** [comment]
That's true.
But I want to change without css editing.

How about this?

1. default : sidebarMenuItemActiveBackground 
2. plugin setting can override default

**Comment by parkerault** [comment]
That sounds like a good solution!

**Comment by basyura** [comment]
I tested some ui theme.
'sidebarMenuItemActiveBackground' is not appropriate.
The document.body's color and backgroundColor apply to Sidetoc pane's style.

I think more better way is apply  reverse document.body's color and backgroundColor
* The document.body.backgroundColor -> sidetoc's hightlight foreground color 
* document.body.color -> sidetoc's highlight background color.

Dark Theme UI mostly ok, but Light Theme UI is not cool. 
(background black, foreground white)

I use Light theme.
I can be support specially only 'Default Light UI' , but does anyone needs this?

If someone is using this plugin conveniently, they will change it to their own favorite color.

**Comment by parkerault** [comment]
Ah, I see the issue. `@sidebarMenuItemActiveBackground` only worked because I was using a dark theme.

The side TOC pane should use the `@noteListBarBackground` color, and if you use `@noteListViewItemActiveBackground` and `@noteListViewItemActiveColor` as defaults, it should match the theme without having to change any settings:

Light Theme:
![image](https://user-images.githubusercontent.com/78856/93724694-22ae2080-fb6f-11ea-8b6e-73f62161a029.png)

Dark Theme:
![image](https://user-images.githubusercontent.com/78856/93724846-5178c680-fb70-11ea-9d04-13823e8683fb.png)

Using extension settings to let people customize the colors is good, but I think by default it should match the theme that the user has chosen.

**Comment by KINGSABRI** [comment]
Hi @parkerault
Any chance to work on making sidetoc to adapt the default theme colors as you've suggested? 
Also, if you can make the sidebar's width resizable? 

Thanks

**Comment by KINGSABRI** [comment]
Hi all, 
I know it's been a while but it seems the only way that worked for to fix the color and width is by editing 

```
~/.config/inkdrop/packages/sidetoc/lib/sidetoc.js
```
Under 
```js
width: {
    title: "side pane width",
    type: "integer",
    default: 200,
},
```

and the color 

```js
highlightColor: {
    title: "highlight color",
    type: "string",
    default: "#C5EAFB",
},
```


Then reload Inkdrop

**Comment by basyura** [comment]
I implemented this feature  in v2.0.0.

**Comment by basyura** [comment]
close
